### ISSUE TITLE
Parity Benchmarking

### ISSUE THREAD

**Issue by deanstef** [drandreaskrueger/chainhammer#5]
Hi all, I am quite impressed from your work here. It is very nice to have such a tool.

I have just a simple question on Parity benchmarking (is the one I am planning to use so far): why you are sending the entire txs workload to just ONE node exposed on port 8545? What if distribute the load on more than one node?
Your actual configuration is with 4 validators (3 + 1 exposed on the "outside"), and then you are loading transaction on such node. This could be a bottleneck.

Moreover, why you do not use one machine for the blockchain network and other fore the workload?

Thank you

**Comment by drandreaskrueger** [comment]
### yes
Yes, all good ideas. For real world applications those might be good ways to increase the TPS even more. Feel free to fork this repo, and just do it. I would be happy to learn what you find out, how much each of your ideas is actually accelerating the whole thing. Ask me anything that might help you doing it.

---

### client software comparison only 

In its first phase chainhammer proved to be fine in this simplified setup - to just compare the software implementations, without e.g. internet bottlenecks, or node load balancing, etc. But for all clients (e.g. quorum, geth, parity), I used _the same setup_: Everything happened on ONE machine. The 3-4 nodes were dockerized, but all docker instances running on the same machine. And my Python hammer scripts too, on the same machine. And it was sending its transactions to the same one node, out of the 3-4 nodes which were the network. Rather identical setup. Study the `run-all_small.sh`and `run.sh` scripts, they call the _different_ network starter bash scripts in the `networks/` folder, and then run the same mostly _identical_ python scripts on each of them.

---

### TODO:

About parity, please see especially these remarks:

* https://github.com/drandreaskrueger/chainhammer/blob/issues/parity10382/docs/FAQ.md#install-on-ubuntu
* https://github.com/paritytech/parity-ethereum/issues/10382#issuecomment-466373932
* https://github.com/paritytech/parity-ethereum/issues/10382#issuecomment-467533391

--> now that the Parity team has finally found a combination of CLI switches that solved that half a year of problems ... I can redo all my runs, to generate new better (faster) diagrams, and higher TPS entries into my tables.   


In short: Eventually an optimized parity turned out to be much faster that the "out of the box" parity, (seemingly roughly as fast as the other client implementations); but it needs that one specific CLI configuration for that.

**Comment by aawaken** [comment]
> --> now that the Parity team has finally found a combination of CLI switches that solved that half a year of problems ... I can redo all my runs, to generate new better (faster) diagrams, and higher TPS entries into my tables.

Did you do new tests? How are the results?

> In short: Eventually an optimized parity turned out to be much faster that the "out of the box" parity (and seemingly roughly as fast as the other client implementations); but for that it needs that one specific combination of CLI switches.

Where can I find the current most efficient node configuration?

Would highly appreciate

**Comment by drandreaskrueger** [comment]
> Did you do new tests for more recent parity versions ?

Not yet, no. Focusing on substrate now. Once that is done, I am going to redo all runs with the then most up-to-date versions of each client.

If urgent for you, feel free to just create those updated numbers. If on the way you run into any client changes, but can solve it, then please submit a pull request. Thanks.

> Where can I find the current most efficient node configuration?

Look [here](https://github.com/paritytech/parity-ethereum/issues/10382#issuecomment-467533391) and above that comment. And at the above comment.
